addons/web_editor/static/lib/odoo-editor/odoo-editor.js

```diff
@@ -4132,6 +4257,118 @@ var exportVariable = (function (exports) {
             }
         }
 
+        // PASTING / DROPPING
+
+        /**
+         * Prepare clipboard data (text/html) for safe pasting into the editor.
+         *
+         * @private
+         * @param {string} clipboardData
+         * @returns {string}
+         */
+         _prepareClipboardData(clipboardData) {
+            const container = document.createElement('fake-container');
+            container.innerHTML = clipboardData;
+            for (const child of [...container.childNodes]) {
+                this._cleanForPaste(child);
+            }
+            return container.innerHTML;
+        }
+        /**
+         * Prepare clipboard data (text/plain) for safe pasting into the editor.
+         *
+         * @private
+         * @param {string} clipboardData
+         * @returns {string}
+         */
+        _prepareTextClipboardData(clipboardData) {
+            const isXML = !!clipboardData.match(/<[a-z]+[a-z0-9-]*( [^>]*)*>[\s\S\n\r]*<\/[a-z]+[a-z0-9-]*>/i);
+            const isJS = !isXML && !!clipboardData.match(/\(\);|this\.|self\.|function\s?\(|super\.|[a-z0-9]\.[a-z].*;/i);
```

Comment:
Another exemple why this doesn't work
```
The Arch-flatterer, with whom all the petty flatterers have intelligence, is a man's self.
 ```
is considered as javascript but
```
<img src="1" onerror="alert('xss')"/>
```
is neither XML nor javascript